stack-graphs/src/cycles.rs

```diff
@@ -173,16 +174,25 @@ impl<A: Appendable + Clone> AppendingCycleDetector<A> {
         result
     }
 
-    pub fn append(
-        &mut self,
+    pub fn append(&mut self, appendables: &mut Appendables<A>, appendage: A) {
+        self.appendages.push_front(appendables, appendage);
+    }
+
+    /// Tests if the path is cyclic. Returns a vector indicating the kind of cycles that were found.
+    /// If appending or concatenating all fragments succeeds, this function will never raise and error.
+    pub fn is_cyclic(
+        &self,
         graph: &StackGraph,
         partials: &mut PartialPaths,
         ctx: &mut A::Ctx,
         appendables: &mut Appendables<A>,
-        appendage: A,
-    ) -> Result<(), PathResolutionError> {
-        let end_node = appendage.end_node(ctx);
-        self.appendages.push_front(appendables, appendage);
+    ) -> Result<Vec<Cyclicity>, PathResolutionError> {
```

Comment:
I think the `Vec` here is indicating that the `Cyclicity` enum values aren't mutually exclusive, and that the result should really be a set.  Is that right?

If so, we might consider a bitset enum here.  Since there are only three values, you'd be able to represent a set in a `u8`.  There are a couple of crates to make this easier (e.g. [`enumset`](https://docs.rs/enumset/latest/enumset/) or [`bitmask_enum`](https://docs.rs/bitmask-enum/latest/bitmask_enum/)).